nautobot/core/utilities.py

```diff
@@ -29,13 +30,19 @@ def check_filter_for_display(filters, field_name, values):
     if field_name not in filters.keys():
         return resolved_filter
 
-    resolved_filter["display"] = filters[field_name].label or field_name
-    if len(values) == 0 or not hasattr(filters[field_name], "queryset") or not is_uuid(values[0]):
+    field = filters[field_name]
+
+    resolved_filter["display"] = (
+        (isinstance(field, RelationshipFilter) and field.relationship.get_label(side=field.side))
+        or field.label
+        or field_name
+    )
```

Comment:
That's a bit of dark magic there relying on `a and b` to return `b` - definitely caused me to do a double-take of "That can't possibly be right, can it?". Maybe clearer even if more verbose:

```suggestion
    resolved_filter["display"] = None
    if isinstance(field, RelationshipFilter):
        resolved_filter["display"] = field.relationship.get_label(side=field.side))
    if not resolved_filter["display"]:
        resolved_filter["display"] = field.label or field_name
```